### ISSUE TITLE
App failed to compile for web

### ISSUE THREAD

**Issue by Shriyanshx** [781flyingdutchman/background_downloader#165]
**Describe the bug**
App compilation failed due to this plugin. 

**To Reproduce**
1. Add plugin to your project
2. run for web (chrome)
3. see error in the logs

**Expected behavior**
App should run and compile on web.

**Logs**
```
Launching lib/main_dev.dart on Chrome in debug mode...
Waiting for connection from debug service on Chrome...
../../../.pub-cache/hosted/pub.dev/background_downloader-7.10.0/lib/src/desktop/isolate.dart:45:36: Error: Member not found: 'BackgroundIsolateBinaryMessenger.ensureInitialized'.
  BackgroundIsolateBinaryMessenger.ensureInitialized(rootIsolateToken);
                                   ^^^^^^^^^^^^^^^^^
```

**flutter doctor**
```
[!] Flutter (Channel stable, 3.13.4, on macOS 14.0 23A339 darwin-arm64, locale en-IN)
    ! Warning: `flutter` on your path resolves to /Users/shriyanshraj/flutter/flutter_3.13/bin/flutter, which is not inside your current Flutter SDK checkout at
      /Users/shriyanshraj/Flutter/flutter_3.13. Consider adding /Users/shriyanshraj/Flutter/flutter_3.13/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /Users/shriyanshraj/flutter/flutter_3.13/bin/dart, which is not inside your current Flutter SDK checkout at
      /Users/shriyanshraj/Flutter/flutter_3.13. Consider adding /Users/shriyanshraj/Flutter/flutter_3.13/bin to the front of your path.
[!] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.
[✓] Xcode - develop for iOS and macOS (Xcode 15.0)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.3)
[✓] VS Code (version 1.82.2)
[✓] Connected device (4 available)
[✓] Network resources

```

**Comment by 781flyingdutchman** [comment]
Sorry to hear that. There must have been some regression from 7.9.4 - can you pin that version and confirm that compiled for web? I should be able to fix this in the next minor update.

**Comment by Shriyanshx** [comment]
@781flyingdutchman, Yes I compiled my app for web, I am using this plugin only on android but it causing issue on running on web.

plugin version - background_downloader: ^7.10.0

**Comment by 781flyingdutchman** [comment]
Thanks, can you try pinning version to 7.9.4

**Comment by 781flyingdutchman** [comment]
Fixed in version 7.10.1
